elastic/kibana
# [RAC] - Update field names

## Summary

Trying to align on field naming. In the code there seems to be the following that are all referencing the context in which a rule instance was created (rules created in stack have consumer of `alerts`):
- `rule.consumers`
- `kibana.alert.consumer`
- `kibana.alert.consumers`
- `kibana.alert.rule.consumers`
- `kibana.alert.owner`

In the code there are also the following all referencing a rule's alert type (ex: `apm.error_rate`):
- `rule.id`
- `kibana.alert.rule.rule_type_id`

It's my understanding that these are all meant to refer to the same thing - the single feature which created the rule instance. So if a rule was created in security solution, the consumer is `siem`, if created in apm it's `apm`. Observability and the rule registry were using `kibana.alert.owner` though it'd been on the list to update to `kibana.alert.consumer` and security solution was using `kibana.alert.owners`. The other ones were there as fields but not being used.

I think there is some confusion as to what `consumer` is meant to denote. It should denote the same thing it does for rules rbac. If there's other context we want to save on an alert, that should be considered a different field, the `consumer` field was meant to be specific to role based access logic shared with alerting framework.

### Fields used moving forward
`kibana.alert.rule.consumer` will refer to the context in which a rule instance is created. Rules created in:
- stack --> `alerts`
- security solution --> `siem`
- apm --> `apm`

`kibana.alert.rule.producer` will refer to the plugin that registered a rule type. Rules registered in:
- stack --> `alerts`
- security solution --> `siem`
- apm --> `apm`

So an `apm.error_rate` rule created in stack will have:
- consumer: `alerts` and producer: `apm`
 An `apm.error_rate` rule created in apm will have:
- consumer: `apm` and producer: `apm`

`kibana.alert.rule.rule_type_id` will refer to a rule's rule type id. Examples:
- `apm.error_rate`
- `siem.signals`
- `siem.threshold`

Also renamed the following because `rule.*` fields are meant to be ecs fields pulled from the source/event document, not refer to our rule fields.
`rule.name` --> `kibana.alert.rule.name` will refer to the rule's name.

`rule.category` --> `kibana.alert.rule.category` will refer to the rule's category.

`rule.id` --> `kibana.alert.rule.uuid` will refer to the rule's uuid.


### To test
Add the following to your `kibana.dev.yml`:
```
xpack.securitySolution.enableExperimental: ['tGridEnabled']
```

Create a custom rule with query `*:*`.

Run `yarn test:generate` within `x-pack/plugins/security_solution`

Observe the magic (alerts should show up in table).

<img width="2556" alt="Screen Shot 2021-08-10 at 10 35 32 PM" src="https://user-images.githubusercontent.com/10927944/128975244-fdcb1750-7d00-4b22-8180-3bb53ce72e3c.png">
